### PR TITLE
Support Record type

### PR DESCRIPTION
**Pull request by yoshihiro503**
Support

- record types
- record field type

See also #54 

## Example
```
-record(state, {name :: string(), param :: term()}).
-type record_as_type(A) :: #state{param :: A}.
```

### CODE REVIEW

File: test/test_type_attr.erl [hunk 1]
```diff
@@ -7,3 +7,6 @@
 
 %% test case for opaque decl
 -opaque int() :: integer().
+
+-record(state, {name :: string(), param :: term()}).
+-type record_as_type(A) :: #state{param :: A}.
```
Comment (amutake) [inline test/test_type_attr.erl]: Could you add the purpose of this test-case as a comment? e.g., `%% test case for record type and record field type`

Comment (yoshihiro503) [reply test/test_type_attr.erl]: I added.

File: lib/abstract_format.ml [hunk 2]
```diff
@@ -1027,6 +1030,16 @@ and type_of_sf sf : (type_t, err_t) Result.t =
      let%bind high = sf_high |> type_of_sf |> track ~loc:[%here] in
      TyRange {line; low; high} |> return
 
+  (* record type : t(A) = #state{name :: A} *)
+  | Sf.Tuple (4, [Sf.Atom "type";
+                  Sf.Integer line;
+                  Sf.Atom "record";
+                  Sf.List (
+                      Sf.Tuple(3, [Sf.Atom "atom"; Sf.Integer line_field_types; Sf.Atom name])
```
Comment (amutake) [inline lib/abstract_format.ml]: `line_field_types` seams to be line of record name. So could you rename `line_field_types` to `line_name`?

Comment (yoshihiro503) [reply lib/abstract_format.ml]: That's right, I fixed it.

Comment (amutake) [reply lib/abstract_format.ml]: thx!

**Pull request merged**